rustls/src/client/hs.rs

```diff
@@ -598,7 +610,6 @@ impl State for ExpectServerHello {
             return Err(illegal_param(sess, "server varied selected ciphersuite"));
         }
 
-        let version = sess.common.negotiated_version.unwrap();
```

Comment:
Why not do this here:
```
sess.common.negotiated_version = Some(version);
```
Instead of twice above?

Reply:
I thought I had this in a previous commit revision but it appears to have been lost somehow. These have now been refactored.